pkg/controller/certificaterequests/sync_test.go

```diff
@@ -180,8 +180,172 @@ func TestSync(t *testing.T) {
+								Type:               cmapi.CertificateRequestConditionDenied,
+								Status:             cmmeta.ConditionTrue,
+								Reason:             "Foo",
+								Message:            "Certificate request has been denied by cert-manager.io",
+								LastTransitionTime: &nowMetaTime,
+							}),
+							gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+								Type:               cmapi.CertificateRequestConditionReady,
+								Status:             cmmeta.ConditionFalse,
+								Reason:             "RequestDenied",
+								Message:            "The CertificateRequest was denied by an approval controller",
+								LastTransitionTime: &nowMetaTime,
+							}),
+							gen.SetCertificateRequestFailureTime(nowMetaTime),
+						),
+					)),
+				},
+			},
+		},
+		"should do nothing if request has both Denied and Ready with reason RequestDenied conditions": {
+			certificateRequest: gen.CertificateRequestFrom(baseCRNotApproved,
+				gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+					Type:               cmapi.CertificateRequestConditionDenied,
+					Status:             cmmeta.ConditionTrue,
+					Reason:             "Foo",
+					Message:            "Certificate request has been denied by cert-manager.io",
+					LastTransitionTime: &nowMetaTime,
+				}),
+				gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+					Type:               cmapi.CertificateRequestConditionReady,
+					Status:             cmmeta.ConditionFalse,
+					Reason:             "RequestDenied",
+					Message:            "The CertificateRequest was denied by an approval controller",
+					LastTransitionTime: &nowMetaTime,
+				}),
+				gen.SetCertificateRequestFailureTime(nowMetaTime),
+			),
+			builder: &testpkg.Builder{
+				CertManagerObjects: []runtime.Object{baseIssuer,
+					gen.CertificateRequestFrom(baseCRNotApproved,
+						gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+							Type:               cmapi.CertificateRequestConditionDenied,
+							Status:             cmmeta.ConditionTrue,
+							Reason:             "Foo",
+							Message:            "Certificate request has been denied by cert-manager.io",
+							LastTransitionTime: &nowMetaTime,
+						}),
+						gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+							Type:               cmapi.CertificateRequestConditionReady,
+							Status:             cmmeta.ConditionFalse,
+							Reason:             "RequestDenied",
+							Message:            "The CertificateRequest was denied by an approval controller",
+							LastTransitionTime: &nowMetaTime,
+						}),
+						gen.SetCertificateRequestFailureTime(nowMetaTime),
+					),
+				},
+				ExpectedEvents:  []string{},
+				ExpectedActions: []testpkg.Action{},
+			},
+		},
+		"should update Failed if certificate request is denied and has Ready Pending condition": {
```

Comment:
`should set Ready condition reason to Failed if certificate request is denied and has a Pending reason for the Ready condition`